### ISSUE TITLE
Test SignalR and ANCM

### ISSUE THREAD

**Issue by anurse** [aspnet/SignalR#131]
We need to do some through testing of SignalR and the ASP.NET Core Module for IIS. The basics should work but there are some known problems and possible unknown problems

* Known: ANCM buffers responses which messes with Server-Sent Events. Solution is to change ANCM to **not** buffer responses with a `Content-Type` of `text/event-stream`
* Unknown: Behavior of ANCM when terminating the connection, does it propagate the TCP FIN to allow the RequestAborted token to fire? This helps with graceful shutdown of SSE connections
* Default Request Timeout is 2mins. This should be fine for Long Polling, but for SSE and WebSockets, we need to make sure we're sending keep-alive frames within the 2min window to keep the connection open.

We need to do some bug bashing to see what we can find.

/cc @shirhatti @moozzyk @davidfowl

**Comment by moozzyk** [comment]
https://github.com/aspnet/AspNetCoreModule/issues/16

**Comment by anurse** [comment]
I did some basic testing with ANCM and found a few things.

1. The RequestAborted token is not fired when the SSE request is cancelled from the end-user. This means ANCM is not properly terminating the underlying connection. However, the token **is** fired when the request times out

1. When an SSE or WS request times out, it is immediately reaped by the ConnectionManager because we never set LastActiveUtc. This may be fine, since the only way an SSE or WS request can terminate this way is if there's an unrecoverable error sending keep-alives

1. We set `Content-Encoding` to `identity` which stops compression from buffering. I confirmed that without this we still get buffering issues. Already talked to @shirhatti about this one, and the issue @moozzyk linked discusses it.
